ml/Host.cc

```diff
@@ -437,6 +477,11 @@ void DetectableHost::detect() {
         updateDetectionChart(getRH(), Dur.count() * 1000);
 
         std::this_thread::sleep_for(Seconds{updateEvery()});
+
+        /*control the size of the database table and shrink them if required*/
+        DB.shrinkAnomalyRateInfoTable(static_cast<int>(Cfg.MaxAnomalyRateInfoDataTableSize));
+        time_t OldestTime = now_realtime_sec() - (Cfg.MaxAnomalyRateInfoDataAge * 3600);
+        DB.removeOldAnomalyRateInfo(OldestTime);
```

Comment:
You are correct -- this is the approach (notice the difference with non dbengine memory mode)

Reply:
OK, reason I believed doing some or all of this control should be done wrt the overall approach, and not as an independent approach.

Reply:
I also believe that we want to remove old data nevertheless, if it is really old, even if we may not insert new data. That also would help saving memory.

Reply:
The period for adding anomaly rates is configurable. Therefore, in case that is set to so long, although it would mean that there won't be a big population of the anomaly rates in the database, yet it doesn't mean that there wasn't a big population from the past, and that could get too old. I think it is more efficient if we take care of the data size regardless if we want to add new data or not. Unless the iterated check of sqlite as often as detect is really costy, and that would come down to the profiling measures, that is on the way to get shape.